test/Npgsql.Tests/Types/CompositeTests.cs

```diff
@@ -408,9 +289,8 @@ public void Recursive()
                 conn.ExecuteNonQuery("CREATE TYPE pg_temp.composite_contained AS (x int, some_text text)");
                 conn.ExecuteNonQuery("CREATE TYPE pg_temp.composite_container AS (a int, contained composite_contained)");
                 conn.ReloadTypes();
-                // Registration in inverse dependency order should work
-                conn.TypeMapper.MapComposite<SomeCompositeContainer>("composite_container");
                 conn.TypeMapper.MapComposite<SomeComposite>("composite_contained");
+                conn.TypeMapper.MapComposite<SomeCompositeContainer>("composite_container");
```

Comment:
Any reason for this order change? It's specifically meant to test that reverse order mapping works, was this depending on unmapped composite support?

Reply:
It worked before because of delayed field mapping, but not now. `CompositeHandlerFactory` fully constructs a factory on request now.

Reply:
Any reason not to retain the delayed field mapping? This forces users to map in the order of dependency (breaking change) and prohibits circular references.

Reply:
PostgreSQL doesn't allow circular references, but this is interesting question. On the one hand, registration order keeps things consistent with PostgreSQL, but on another there is user experience.

Reply:
I don't think it's super critical - if there's a good reason we can't support it, then OK. But if there isn't, then it's better to have it...